src/modules/external/aha/components/AhaScore.tsx

```diff
@@ -0,0 +1,134 @@
+import { Stack, Typography } from '@mui/material';
+import { useMemo, useState } from 'react';
+import LabelWithContent from '@/components/elements/LabelWithContent';
+import LoadingButton from '@/components/elements/LoadingButton';
+import ApolloErrorAlert from '@/modules/errors/components/ApolloErrorAlert';
+import ErrorAlert from '@/modules/errors/components/ErrorAlert';
+import { emptyErrorState, ErrorState, hasErrors } from '@/modules/errors/util';
+import { ChangeType, GroupItemComponentProps } from '@/modules/form/types';
+import { useFetchAhaScoreMutation } from '@/types/gqlTypes';
+
+const SCORE_LINK_ID = 'aha_score';
+const ALT_AHA_FLAG_LINK_ID = 'aha_alt_aha_flag';
+const DW_CLIENT_LINK_ID = 'aha_dw_client_id';
+const GENERATOR_LINK_ID = 'aha_generator';
+
+const EXPECTED_LINK_IDS = [
+  SCORE_LINK_ID,
+  ALT_AHA_FLAG_LINK_ID,
+  DW_CLIENT_LINK_ID,
+  GENERATOR_LINK_ID,
+];
+
+// Custom component for rendering AHA score. Renders a button to fetch the score, as well as sub-items including:
+// score (value), alt-AHA flag, and hidden items for values that should be submitted but not shown to the user.
+// Similar to DisabilityGroup, this component is tightly coupled to its expected form structure.
+const AhaScore = ({
+  item,
+  handlers,
+  renderChildItem,
+  severalItemsChanged,
+}: GroupItemComponentProps) => {
+  // Introspect on the group's child elements and ensure items exist for the expected link IDs
+  const isComponentValid = useMemo(() => {
+    return EXPECTED_LINK_IDS.every((expectedLinkId) => {
+      return item.item?.find((i) => i.linkId === expectedLinkId);
+    });
+  }, [item]);
+
+  const [errorState, setErrorState] = useState<ErrorState>(emptyErrorState);
+
+  // hasScore is a boolean that can be undefined when the user hasn't clicked the fetch button yet.
+  // `false` indicates the user did click the button but the score wasn't available.
+  const [hasScore, setHasScore] = useState<boolean | undefined>(undefined);
```

Comment:
Just to make sure I'm understanding this, is it a combination of "has score" and "has fetched" eg
undefined= haven't fetched yet
true = fetched and found score
false = fetched and didn't find score
?

I think using a boolean state with 3 states is making the render logic kind of hard to follow. For future readability/maintainability, I'd suggest splitting it out into 2 states (has fetched and has score). Or you can rely on the [called](https://www.apollographql.com/docs/react/data/mutations#result) result value on the mutation to know whether the mutation has been called.